src/__utils/Collapse.tsx

```diff
@@ -0,0 +1,45 @@
+import React from "react";
+
+export interface CollapseProps extends React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement> {
+    /** True to expand, false to collapse */
+    toggle?: boolean;
+}
+
+export const Collapse: React.FC<CollapseProps> = React.memo(({ toggle = false, ...props }: CollapseProps) => {
+    const collapseRef: React.MutableRefObject<HTMLDivElement> = React.useRef<HTMLDivElement>();
+    const [height, setHeight] = React.useState<"auto" | number>(0);
```

Comment:
Optional: I think you can use React.StyleAttributes<HTMLDivElement>["style"]["height"] and ["display"]

Reply:
Tried it, didn't work, however, it's unnecessary since this is used internally and I'm only defining the values that I will be using.